frame/staking/src/pallet/mod.rs

```diff
@@ -1038,7 +1054,9 @@ pub mod pallet {
 						}
 					})
 				})
-				.collect::<result::Result<Vec<T::AccountId>, _>>()?;
+				.collect::<result::Result<Vec<T::AccountId>, _>>()?
+				.try_into()
+				.expect("bound checked above; qed");
```

Comment:
Why cant this also just return an error of `TooManyTargets`?

Reply:
could, but once I finish something like `.defensive_map_err()`, because it really cannot ever happen.

Reply:
I would much rather you return a result here than have an expect. You never know how refactoring or whatever happens. Lets just not use an expect where we clearly do not need it.